addons/mail/models/discuss/discuss_channel.py

```diff
@@ -294,9 +294,10 @@ def write(self, vals):
         if 'image_128' in vals:
             for channel in self:
                 notifications.append([channel, 'mail.record/insert', {
-                    'Channel': {
+                    'Thread': {
```

Comment:
For me the diff in this file is a pure regression, and the desired result is what we had before.
But it's acceptable if it allows moving forward with the cleaning, until we are ready to make the change again.

Reply:
I don't get what's the regression in this file. We reuse `Thread` concept and we should add `model: "discuss.channel"`? This is far better than have another other concept `Channel` that share the same shape as `Thread` except the model is optional, and it has some new fields for itself, and some fields that doesn't work and are chatter-specific.

As I said in another comment, differences between channels and chatters come essentially from poor implementation choices, not conceptual differences. Sure there are some functional choices for differences, but they both want users to interact with chat features. Stating that there are more differences than commonalities between both models is ridiculous, and the less concepts the better. Thread models a conversation, adding a sub-model has no value and more drawbacks on code maintenance.